spherical_geometry/polygon.py

```diff
@@ -1166,12 +1166,30 @@ def multi_union(cls, polygons):
 
         for polygon in polygons:
             if not isinstance(polygon, SphericalPolygon):
-                raise TypeError
+                raise TypeError("Expected a sequence of SphericalPolygon")
+
+        # Next block is a workaround to a bug in the graph code that leads to a
+        # crash when computing multi_union of polygons
+        # when some of input polygons are very close to each other.
+        # Remove the next block once the bug is fixed.
+        # See https://github.com/spacetelescope/spherical_geometry/issues/232
+        points = [np.sort(list(polygons[0].points)[0], axis=0)]
```

Comment:
If I understand this correctly, points is really a list of polygons as defined by their vertices. Seems like it should be polygons or something that indicates that. I find it a bit confusing for that reason. Or it may be my hotspot induced frustration.

Reply:
> If I understand this correctly, points is really a list of polygons as defined by their vertices. Seems like it should be polygons or something that indicates that. I find it a bit confusing for that reason. Or it may be my hotspot induced frustration.

I do not understand this comment at all. Sorry! Could you, please, reformulate? Specifically, I am confused by this part:

> points is really a list of polygons as defined by their vertices. Seems like it should be polygons or something that indicates that.

points is a list of lists of 3 values (`x`, `y`, and `z`) - the coordinates of the vertices. Of course vertices define a polygon but I do not see an issue with that ("Seems like it should be polygons").

Reply:
My comment is about the variable `points,` which really is a list of the points for each polygon that passes the filter. In the enclosed loop, your loop variable is `p`, from which a set of points is obtained, but `p` represents a polygon. In other words, the outer loop is taking members from a variable called points, which is being compared to the inner loop which is looping over polygons. Seems to me that the outer loop is also looping over polygons (the members of which are the points of the polygon). So yes, one list doesn't contain polygon objects, but still the members are essentially polygons. Does the name points mean the points of a polygon, or a list of points of each polygon (the latter I presume)? It just seems a confusing use of the term points. The contents of points really are the lists of polygon vertices and why not refer to that as a list of the polygons, with only the vertices representing them. Maybe something like `filtered_vpolygons `(vertex polygons?). It is basically a different representation than `polygons` uses, right?

Reply:
> So yes, one list doesn't contain polygon objects, but still the members are essentially polygons.

If I need polygon vertices (`==points`) the I will call them points (as `spherical_geometry` defined them from inception). This is distinguished from the `polygon.SphericalPolygon` _object_. True, vertices almost define a polygon but they are still distinct concepts. I disagree with _"but still the members are essentially polygons"_. No, they are points.

> Does the name points mean the points of a polygon, or a list of points of each polygon (the latter I presume)? It just seems a confusing use of the term points.

The later.

> The contents of points really are the lists of polygon vertices and why not refer to that as a list of the polygons.

Because polygons are not vertices.

Reply:
I renamed `points->accepted_polygon_points`.

Reply:
OK. I'm not sure that it conveys that these are grouped polygon, but I lack the will to continue :-)